doc/examples/server_selection.rst

```diff
@@ -0,0 +1,82 @@
+Server Selector Example
+=======================
+
+Users can exert fine-grained control over the `server selection algorithm`_
+by setting the `serverSelector` option on the :class:`~pymongo.MongoClient`
+to an appropriate callable. This example shows how to use this functionality
+to prefer servers running on ``localhost``.
+
+
+.. note::
+
+   Directing queries to servers running on ``localhost`` can be desirable
+   when using a sharded cluster with multiple ``mongos``, as locally run
+   queries are likely to see lower latency and higher throughput.
```

Comment:
Can you replace the note with a ".. warning::" that says this is a power user feature? Misusing a custom server selectors may lead to serious issues such as heart attack or stroke. Please consult your doctor before attempting to use server selectors. etc...

The existing note can be moved to a normal paragraph in the "Example: ..." section below. Also please mention that it's highly dependent on the application if preferring localhost is beneficial or not.